factory/README.md

```diff
@@ -0,0 +1,172 @@
+* edge
+* cypress
+
+## Benefits
+
+* Freedom to choose which versions to test against.
+* No need to wait on an official release to test the latest version of a browser.
+* Smaller docker sizes especially when not including unused browsers.
+* Easily test multiple browser versions.
+* Reduced maintenance and pull requests for the cypress-docker repo.
+* Ability to offer more variations of docker containers at low cost.
+
+## Usage
+
+The cypress docker factory works by relying on the `ONBUILD` docker command to run commands at build time of the the consuming container. `Args` can be specified in a number of ways, some of which are demonstrated below. For each of these examples we are building the equivalent of the `cypress/browsers` docker container. In each instance since the example is only testing the chrome version, the examples could just install chrome by itself if the other browsers were not used.
+
+### In the dockerfile
+
+Args can be defined directly in the dockerfile.
+
+dockerfile
+
+```dockerfile
+# Args are defined in the dockerfile before the FROM command.
+# Using these args will cause an image to be created with node, chrome, firefox and edge.
+ARG CHROME_VERSION='107.0.5304.121-1'
+ARG EDGE_VERSION='100.0.1185.29-1'
+ARG FIREFOX_VERSION='107.0'
+
+FROM cypress/factory
+
+COPY . /opt/app
+WORKDIR /opt/app
+RUN npm install --save-dev cypress
+```
+
+build commands
+
+```bash
+docker build . -t test
+docker run -it test npm run test -b chrome
+```
+
+### At build time
+
+Args can be passed to the docker build command.
+
+dockerfile
+
+```dockerfile
+FROM cypress/factory
+
+COPY . /opt/app
+WORKDIR /opt/app
+RUN npm install --save-dev cypress
+```
+
+build commands
+
+```bash
+docker build . --build-arg CHROME_VERSION='107.0.5304.121-1' --build-arg EDGE_VERSION='100.0.1185.29-1' --build-arg FIREFOX_VERSION='107.0' -t test
+
+docker run test npm run test -b chrome
+```
+
+### In docker-compose.yml
+
+Finally, args can be specified in the docker-compose.yml file.
+
+docker-compose.yml
+
+```yml
+version: '3'
+
+services:
+  test:
+    args:
+      CHROME_VERSION: '107.0.5304.121-1'
+      EDGE_VERSION: '100.0.1185.29-1'
+      FIREFOX_VERSION: '107.0'
+    build:
+      context: .
+    command: npm run test
+```
+
+dockerfile
+
+```dockerfile
+FROM cypress/factory
+
+COPY . /opt/app
+WORKDIR /opt/app
+RUN npm install --save-dev cypress
+```
+
+docker commands
+
+```bash
+docker-compose build test
+
+docker-compose run test
+```
+
+## API
+
+The following args can be set to define what versions the cypress factory docker includes in it's final build.
+
+If no args are defined, only the default version of node will be installed. This can still be a useful container though since we will also install any required cypress apt dependencies.
+
+### NODE_VERSION
+
+The version of Node to install in the docker container. If not specified, the default version of Node (defined [here](./docker-compose.yml)) is installed. Node is required.
```

Comment:
Should the link be to the .env file?  Also, will this typically match a cypress minimum version or how are we bumping this?

Reply:
oh, yes. it used to be defined in docker-compose but i moved it.

Reply:
It seems like it should typically match the bundled node version with cypress, but i'm not sure thats currently the case.

Reply:
Matching the bundled version seems safe. Assume this means we'll likely release a MVB of the factory image when we release a new cypress major version?

Reply:
yeah, we'd have to

Reply:
if we wanted to keep the node version pegged.

Reply:
at a minimum i think we'd release a major version of factory if we drop support for the default node version.

Reply:
we could always build base/browsers/included with a different node version than the default.